src/platform/Ameba/ConfigurationManagerImpl.cpp

```diff
@@ -152,6 +152,30 @@ CHIP_ERROR ConfigurationManagerImpl::GetPrimaryWiFiMACAddress(uint8_t * buf)
     return CHIP_NO_ERROR;
 }
 
+CHIP_ERROR ConfigurationManagerImpl::GetUniqueId(char * buf, size_t bufSize)
+{
+#ifdef CHIP_DEVICE_CONFIG_UNIQUE_ID
+    ReturnErrorCodeIf(bufSize < sizeof(CHIP_DEVICE_CONFIG_UNIQUE_ID), CHIP_ERROR_BUFFER_TOO_SMALL);
+    strcpy(buf, CHIP_DEVICE_CONFIG_UNIQUE_ID);
+    return CHIP_NO_ERROR;
+#else
+    ReturnErrorCodeIf(bufSize != 32, CHIP_ERROR_INVALID_MESSAGE_LENGTH);
+    char temp[32];
+    int i = 0, j = 0;
+
+    memset(&temp[0], 0, 32);
+    ReturnErrorCodeIf(wifi_get_mac_address(temp) != 0, CHIP_ERROR_INVALID_ARGUMENT);
```

Comment:
The specification says that UniqueID may be constructed e.g. from a MAC address, but in order to prevent tracking, it cannot be used as-is. Moreover, it should be updated (changed) after factory reset (maybe it's fine for MAC if it's changing, but `CHIP_DEVICE_CONFIG_UNIQUE_ID` above does not change.

Reply:
Hi @LuDuda,
Thanks for the reminder, the CHIP_DEVICE_CONFIG_UNIQUE_ID is currently set as a fix string, mainly for test and verification purpose for now. This shall be revised in future to make it more systematic and tracking prevented.